### PR TITLE
-- added more timing to get better coverage of cpr solver

### PR DESCRIPTION
**Pull request by hnil**
-- added timing macros
-- added include files need
This give nearly full coverage of the cpr solver forexample for tracy profiling.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/simulators/linalg/amgcpr.hh [hunk 1]
```diff
@@ -841,6 +851,7 @@ namespace Dune
 
     template<class M, class X, class S, class PI, class A>
     void AMGCPR<M,X,S,PI,A>::mgc(LevelContext& levelContext){
+      //OPM_TIMEBLOCK(mgc);
```
Comment (blattms) [inline opm/simulators/linalg/amgcpr.hh:854]: Is this commented out intentionally

**Comment by blattms** [comment]
jenkins build this please

File: opm/simulators/linalg/amgcpr.hh (hunk 1 above)
Comment (hnil) [reply opm/simulators/linalg/amgcpr.hh:854]: yes. Some of the macros have troubles if functions are to simple. Scope probably is removed ??

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: drogon - Threads: 1                | 1.032    |
| opm-git | OPM Benchmark: drogon - Threads: 8                | 0.855    |
| opm-git | OPM Benchmark: punqs3 - Threads: 1                | 0.98     |
| opm-git | OPM Benchmark: punqs3 - Threads: 8                | 0.98     |
| opm-git | OPM Benchmark: smeaheia - Threads: 1              | 0.989    |
| opm-git | OPM Benchmark: smeaheia - Threads: 8              | 0.878    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 1         | 1.012    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 8         | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.904    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.997    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.938    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.982    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.973    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=2175

**Review by atgeirr — APPROVED** [review]

**Pull request merged**